scripts/system/create/entityProperties/html/js/entityProperties.js

```diff
@@ -4373,6 +4408,16 @@ function handleEntitySelectionUpdate(selections, isPropertiesToolUpdate) {
             hideMaterialDataSaved();
         }
 
+        var materialDataText = JSON.stringify(materialJson);
+        if (getPropertyInputElement("materialURL").value === "materialData" && materialDataText.indexOf("hifi_shader_simple") === -1 &&
+                materialDataText.indexOf("glossMap") === -1 && materialDataText.indexOf("specularMap") === -1 && materialDataText.indexOf("bumpMap") === -1 && 
+                materialDataText.indexOf("lightMap") === -1  && materialDataText.indexOf("texCoordTransform0") === -1 && materialDataText.indexOf("texCoordTransform1") === -1 &&
+                (materialJson.materials === undefined || materialJson.materials.length <= 1 || typeof materialJson.materials === "object")) {
+            showMaterialAssistantButton();
+        } else {
+            hideMaterialAssistantButton();
+        }
+
```

Comment:
Would it be worth putting this into a function and reusing it in the same/similar code circa line 3530?

Reply:
I was doing that when I saw your comments.
Solved. Availabilitty conditions has been centralized in materialAssistant.js (whit other functions related to the Material Assistant)